script/DeployRehearsalContracts.s.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Deployer} from "@eth-optimism-bedrock/scripts/Deployer.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import {ProxyAdmin} from "@eth-optimism-bedrock/src/universal/ProxyAdmin.sol";
+import {Proxy} from "@eth-optimism-bedrock/src/universal/Proxy.sol";
+import {Script} from "forge-std/Script.sol";
+
+import { console2 as console } from "forge-std/console2.sol";
+
+// forge script scripts/Deploy.s.sol:Deploy --private-key $PRIVATE_KEY --broadcast --rpc-url $ETH_RPC_URL
+// forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --private-key $PRIVATE_KEY --broadcast --rpc-url $ETH_RPC_URL
+contract DeployRehearsalContracts is Deployer {
+
+    GnosisSafe owner_safe;
+    GnosisSafe council_safe;
+    GnosisSafe foundation_safe;
+    ProxyAdmin proxy_admin;
+
+    /// @notice The name of the script, used to ensure the right deploy artifacts
+    ///         are used.
+    function name() public pure override returns (string memory name_) {
+        name_ = "DeployRehearsalContracts";
+    }
+
+    function setUp() public override {
+        super.setUp();
+
+        owner_safe = GnosisSafe(payable(vm.envAddress("OWNER_SAFE")));
+        council_safe = GnosisSafe(payable(vm.envAddress("COUNCIL_SAFE")));
+        foundation_safe = GnosisSafe(payable(vm.envAddress("FOUNDATION_SAFE")));
+
+        require(owner_safe.isOwner(address(council_safe)));
+        require(owner_safe.isOwner(address(foundation_safe)));
+
+        console.log("Deploying from %s", deployScript);
+        console.log("Deployment context: %s", deploymentContext);
+    }
+
+    function run() public {
```

Comment:
This is deploying the new system which will be upgraded?

Reply:
Assuming it is, not initializing the Proxys prior to the rehearsal will likely make the tenderly simulation diff larger than it would otherwise be.